### ISSUE TITLE
Fix false negatives for styled-components CSS-in-JS

### ISSUE THREAD

**Issue by ai** [stylelint/stylelint#3690]
> Clearly describe the bug

Stylelint CLI can’t find errors in CSS in JS files.

> What CSS is needed to reproduce the bug

```js
const Component = styled.div`
  margin-top: 10px;
  margin: 0 auto;
`
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

> Which version of stylelint are you using?

`9.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

CSS-in-JS

> What did you expect to happen?

It should find an error in the `Component`

> What actually happened (e.g. what warnings or errors did you get)?

Nothing

**Comment by ai** [comment]
In the same project this file:

```css
.component {
  margin-top: 10px;
  margin: 0 auto;
}
```

returns errors

```
➤ n stylelint test.css 

test.css
 3:3  ✖  Unexpected           declaration-block-no-shorthand-property-overrides
         shorthand "margin"                                                    
         after "margin-top"
```

**Comment by jeddy3** [comment]
@ai Thanks for the report and for using the template.

I am unable to reproduce this locally, both with `9.5.0` and `master`.

The following:

```js
// styled.js
const Component = styled.div`
  margin-top: 10px;
  margin: 0 auto;
`
```

Gives me:

```console
jeddy3:stylelint-test jeddy3$ ./node_modules/.bin/stylelint "sources/js/styled.js" --config configs/styled.json

sources/js/styled.js
 4:3  ✖  Unexpected shorthand "margin" after         declaration-block-no-shorthand-property-overrides
         "margin-top"
```

Are you able to create a reproducible test repo to help us get to the bottom of this, please?

**Comment by ai** [comment]
Hm, I will look again on Monday. Maybe there is something wrong with our config.

**Comment by ai** [comment]
@jeddy3 I found the problem. It happens because I used `css-literal-loader` instead of `styled-components`. It has the similar API (but much better :D).

**Comment by ai** [comment]
I added issue to correct repo https://github.com/gucong3000/postcss-jsx/issues/36
